codec-dns/src/main/java/io/netty/handler/codec/dns/DefaultDnsRecordDecoder.java

```diff
@@ -90,6 +94,57 @@ protected DnsRecord decodeRecord(
             String name, DnsRecordType type, int dnsClass, long timeToLive,
             ByteBuf in, int offset, int length) throws Exception {
 
+        if (type == DnsRecordType.A) {
+            byte[] buf = new byte[length];
+            in.getBytes(offset, buf);
+            return new DefaultDnsARecord(
+                    name, dnsClass, timeToLive, (Inet4Address) InetAddress.getByAddress(buf));
+        }
+        if (type == DnsRecordType.AAAA) {
+            byte[] buf = new byte[length];
+            in.getBytes(offset, buf);
+            return new DefaultDnsAAAARecord(
+                    name, dnsClass, timeToLive, (Inet6Address) InetAddress.getByAddress(buf));
+        }
+        if (type == DnsRecordType.CNAME) {
+            return new DefaultDnsCNameRecord(
+                    name, dnsClass, timeToLive, decodeName0(in.duplicate().setIndex(offset, offset + length)));
+        }
+        if (type == DnsRecordType.MX) {
+            ByteBuf buf = in.retainedDuplicate().setIndex(offset, offset + length);
```

Comment:
Check if the content is at least longer than 2 bytes?

Reply:
done